EventFilter/HcalRawToDigi/plugins/HcalDigiToRawuHTR.cc

```diff
@@ -0,0 +1,297 @@
+  const HcalElectronicsMap* readoutMap = item.product();
+
+  //collection to be inserted into event
+  std::auto_ptr<FEDRawDataCollection> fed_buffers(new FEDRawDataCollection());
+  
+  //
+  //  Extracting All the Collections containing useful Info
+  iEvent.getByToken(tok_QIE10DigiCollection_,qie10DigiCollection);
+  const QIE10DigiCollection& qie10dc=*(qie10DigiCollection);
+  iEvent.getByToken(tok_QIE11DigiCollection_,qie11DigiCollection);
+  const QIE11DigiCollection& qie11dc=*(qie11DigiCollection);
+  iEvent.getByToken(tok_HBHEDigiCollection_,hbheDigiCollection);
+  const HBHEDigiCollection& qie8hbhedc=*(hbheDigiCollection);
+  iEvent.getByToken(tok_HFDigiCollection_,hfDigiCollection);
+  const HFDigiCollection& qie8hfdc=*(hfDigiCollection);
+  iEvent.getByToken(tok_TPDigiCollection_,tpDigiCollection);
+  const HcalTrigPrimDigiCollection& qietpdc=*(tpDigiCollection);
+
+  // first argument is the fedid (minFEDID+crateId)
+  map<int,HCalFED*> fedMap;
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  // QIE10 precision data
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  uHTRpacker uhtrs;
+  // loop over each digi and allocate memory for each
+
+  for (unsigned int j=0; j < qie10dc.size(); j++){
+    QIE10DataFrame qiedf = static_cast<QIE10DataFrame>(qie10dc[j]);
+    DetId detid = qiedf.detid();
+    HcalElectronicsId eid(readoutMap->lookup(detid));
+    int crateId = eid.crateId();
+    int slotId = eid.slot();
+    int uhtrIndex = ((slotId&0xF)<<8) | (crateId&0xFF);
+
+    /* Defining a custom index that will encode only
+       the information about the crate and slot of a 
+       given channel:   crate: bits 0-7
+                        slot:  bits 8-12 */
+
+    if( ! uhtrs.exist( uhtrIndex ) ){
+      uhtrs.newUHTR( uhtrIndex );
+    }
+    uhtrs.addChannel(uhtrIndex,qiedf,_verbosity);
+  }
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  // QIE11 precision data
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  //uHTRpacker uhtrs;
+  // loop over each digi and allocate memory for each
+  for (unsigned int j=0; j < qie11dc.size(); j++){
+    QIE11DataFrame qiedf = static_cast<QIE11DataFrame>(qie11dc[j]);
+    DetId detid = qiedf.detid();
+    HcalElectronicsId eid(readoutMap->lookup(detid));
+    int crateId = eid.crateId();
+    int slotId = eid.slot();
+    int uhtrIndex = ((slotId&0xF)<<8) | (crateId&0xFF);
+
+    if( ! uhtrs.exist(uhtrIndex) ){
+      uhtrs.newUHTR( uhtrIndex );
+    }
+    uhtrs.addChannel(uhtrIndex,qiedf,_verbosity);
+  }
+
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  // HF (QIE8) precision data
+  // - - - - - - - - - - - - - - - - - - - - - - - - - - - 
+  // loop over each digi and allocate memory for each
+  for(HFDigiCollection::const_iterator qiedf=qie8hfdc.begin();qiedf!=qie8hfdc.end();qiedf++){
+    //HFDataFrame qiedf = static_cast<HFDataFrame>(qie8hfdc[j]);
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant